src/components/RequireSubmission.js

```diff
@@ -10,11 +13,23 @@ import { Redirect } from 'react-router-dom';
  * If there is no submission, the user is forcibly redirected to the start of the form.
  */
 const RequireSubmission = ({ submission, component: Component, ...props }) => {
+  const maintenanceMode = props?.form?.maintenanceMode;
+  const isFormDesigner = IsFormDesignerHeader.getValue();
+  let maintenanceModeAlert = null;
+  if (!isFormDesigner && maintenanceMode) {
+    throw new ServiceUnavailable('Service Unavailable', 503, 'Form in maintenance', 'form-maintenance');
+  } else if (isFormDesigner && maintenanceMode) {
+    maintenanceModeAlert = <MaintenanceMode />;
+  }
+
   if (!submission || !Object.keys(submission).length) {
     return (<Redirect to="/" />);
   }
   return (
-    <Component submission={submission} {...props} />
+    <>
+      {maintenanceModeAlert}
+      <Component submission={submission} {...props} />
+    </>
```

Comment:
this is not ideal, I'd rather have it under the title, but that messes with the structure/re-usability of components so I took the path of least resistance.